src/services/Sso.php

```diff
@@ -0,0 +1,186 @@
+                    $userName     = Craft::$app->getUser()->getIdentity()->username;
+                    $RequestUrl   = $forumUrl."/register/setauthtoken";
+                    $postData     = array('type'=>'json','apikey' => $forumApiKey, 'user' => $userName,'email'=>$userEmail,'externalUserid'=>$userId);
+                    $result       = Websitetoolboxforum::getInstance()->sso->sendApiRequest('POST',$RequestUrl,$postData,'json'); 
+                    setcookie("forumLogoutToken", $result->authtoken, time() + 3600,"/");
+                    setcookie("forumLoginUserid", $result->userid, time() + 3600,"/");
+              }
+          }         
+     }
+     function afterUserCreate($userId){ 
+        $forumUrl     = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumUrl',false);
+        $forumApiKey  = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumApiKey',false); 
+        $postData     = array( 
+                          'type'=>'json',
+                          'apikey'          => $forumApiKey,
+                          'member'          => $_POST['username'],
+                          'externalUserid'  => $userId, 
+                          'email'           => $_POST['email']);
+        if(isset($_POST['firstName'])){
+           $postData['name']  =  $_POST['firstName'];
+        }
+        if(isset($_POST['firstName'])){
+           $postData['name'] .=  " ".$_POST['lastName'];
+        }        
+        $RequestUrl           = $forumUrl . "/register/create_account/";
+        $result               = Websitetoolboxforum::getInstance()->sso->sendApiRequest('POST',$RequestUrl,$postData,'json');          
+    }
+    function afterUpdateUser(){
+      $emailToVerify  = $_SESSION['userEmailBeforeUpdate'];
+      $userId         = Websitetoolboxforum::getInstance()->sso->getUserid($emailToVerify);
+      $userName       = $_POST['username'];
+      $externalUserid = $_POST['userId'];
+      $email          = $_POST['email'];;
+      $userDetails    = array(
+                          "type"=>"json",
+                          "email"          => $email,
+                          "username"       => $userName,
+                          "externalUserid" => $externalUserid,
+                          "name"           => $userName);
+      $url            = WT_API_URL ."/users/$userId";
+      $response       = Websitetoolboxforum::getInstance()->sso->sendApiRequest('POST',$url,$userDetails,'json','forumApikey');
+    }
+    function getUserid($userEmail){         
+         if ($userEmail) {
+            $data     = array(
+                           "email" => $userEmail);
+            $url      = WT_API_URL . "/users/";
+            $response = Websitetoolboxforum::getInstance()->sso->sendApiRequest('GET', $url, $data,'json','forumApikey');              
+            if ($response->{'data'}[0]->{'userId'}) {
```

Comment:
please add a check of "$response" in the condition because in case of userid didn't return then show a warning.